sdk/storage/storage-blob/test/utils/index.ts

```diff
@@ -158,19 +160,36 @@ export async function createRandomLocalFile(
 export async function createRandomLocalFile(
   folder: string,
   blockNumber: number,
-  blockSizeOrContent: number | Buffer
+  blockSize: number,
+  totalSize: number
+): Promise<string>;
+export async function createRandomLocalFile(
+  folder: string,
+  blockNumber: number,
+  blockSizeOrContent: number | Buffer,
+  totalSize?: number
 ): Promise<string> {
   return new Promise<string>((resolve, reject) => {
     const destFile = path.join(folder, getUniqueName("tempfile."));
     const ws = fs.createWriteStream(destFile);
     let offsetInMB = 0;
+    let lastBlockSize = blockSizeOrContent as number;
 
-    function randomValueHex() {
+    if (totalSize !== undefined) {
+      const blockSize = blockSizeOrContent as number;
+      blockNumber = Math.ceil(totalSize / blockSize);
+      lastBlockSize = totalSize - (blockNumber - 1) * blockSize;
```

Comment:
>totalSize [](start = 22, length = 9)

totalSize may less than (blockNumber - 1) * blockSize

Reply:
blockNumber is ignored when totalSize is specified.

Reply:
Then `blockNumber` should not be a required parameter when `totalSize` provided.

---
In reply to: [489273082](https://github.com/Azure/azure-sdk-for-js/pull/11195#discussion_r489273082) [](ancestors = 489273082)

Reply:
Fixed.